charts/victoria-metrics-anomaly/templates/pod-monitor.yaml

```diff
@@ -1,4 +1,4 @@
-{{- if .Values.podMonitor.enabled -}}
+{{- if and .Values.podMonitor.enabled (((.Values.config).monitoring).pull).port -}}
```

Comment:
Maybe it's better to throw an error in this case?
While disabling `PodMonitor` will make sure there are no scrape errors it will make it harder for user to find an error. Explicitly failing here will give user more feedback and enforce using correct configuration.

Reply:
changed this. also added a check for `monitoring.pull.enabled property`